controllers/twophase/state_machine.go

```diff
@@ -0,0 +1,173 @@
+// Copyright 2019 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package twophase
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+
+	"github.com/chaos-mesh/chaos-mesh/api/v1alpha1"
+	"github.com/chaos-mesh/chaos-mesh/pkg/utils"
+)
+
+type chaosStateMachine struct {
```

Comment:
how about adding unit tests for the `chaosStateMachine `

Reply:
@cwen0 There are 6 * 6 = 36 different branches (though most of them are duplicated). I don't think drawing a clear diagram is possible :crying_cat_face: .

Reply:
`unit test` has been added